packages/tracing/src/browser/metrics.ts

```diff
@@ -175,6 +208,68 @@ export class MetricsInstrumentation {
           po.takeRecords().forEach(updateLCP);
         }
       };
+
+      document.addEventListener(
+        'visibilitychange',
+        () => {
+
+          if (this._lcpFinal) {
+            return;
+          }
+
+          if (document.visibilityState === 'hidden') {
+            if (po.takeRecords) {
+              po.takeRecords().forEach(updateLCP);
+              this._lcpFinal = true;
+            }
+          }
+
+        },
+        { capture: true, once: true },
+      );
+    } catch (e) {
+      // Do nothing if the browser doesn't support this API.
+    }
+  }
+
+  /** Starts tracking the First Input Delay on the current page. */
+  private _trackFID(): void {
+    // Based on reference implementation from https://web.dev/fid/#measure-fid-in-javascript.
+    // Use a try/catch instead of feature detecting `first-input`
+    // support, since some browsers throw when using the new `type` option.
+    // https://bugs.webkit.org/show_bug.cgi?id=209216
+    try {
+      if (!PerformanceObserver.supportedEntryTypes.includes('first-input')) {
+        return;
+      }
+
+      const updateFID = (entry: PerformanceEventTiming, po: PerformanceObserver): void => {
+        // Only report FID if the page wasn't hidden prior to
+        // the entry being dispatched. This typically happens when a
+        // page is loaded in a background tab.
+        if (entry.startTime < this._firstHiddenTime) {
+          const fidValue = entry.processingStart - entry.startTime;
+          const timeOrigin = msToSec(performance.timeOrigin);
+          const startTime = msToSec(entry.startTime as number);
+          logger.log('[Measurements] Adding FID');
+          this._measurements['fid'] = { value: fidValue };
+          this._measurements['mark.fid'] = { value: timeOrigin + startTime };
+
+          po.disconnect();
```

Comment:
Should we always disconnect, outside of the if block?!

Reply:
Done in https://github.com/getsentry/sentry-javascript/pull/2909/commits/d5f0a5e053f0c1a8e318fcb7cb3c1b00f9426569